### DISCUSSION TITLE
Facing use while declare custom Lookup field

### DISCUSSION THREAD

**Discussion by he-kishor** [encode/django-rest-framework#9457] [Question & Answer]
I am facing an issue while using lookup_field. I declared it as email, but when I request the method, it shows "not found". The email field has a unique constraint. However, when I use pk and search using pk, it shows me the user data.

views code is given below
class UserViewSetTesting(viewsets.ViewSet):
    """A  simple ViewSet for listing or retreiving users"""
    lookup_field = 'email'
    def list(self,request):
        queryset=User.objects.all()
        serializer=UserSerializers(queryset, many=True)
        return Response(serializer.data)
    
    def retrieve(self,request,email=None):
        queryset=User.objects.all()
        user=get_object_or_404(queryset,email=email)
        serializer=UserSerializers(user)
        return Response(serializer.data)
    the get method URL is 
http://127.0.0.1:8000/api/usertesting/hemant272017@gmail.com
shows me not found while the email Id is correct

**Comment by stabldev** [comment]
hey @he-kishor, could you try this: https://stackoverflow.com/a/55763166

**Reply by he-kishor** [reply]
sure
